hbase-server/src/main/java/org/apache/hadoop/hbase/tool/BulkLoadHFilesTool.java

```diff
@@ -768,13 +782,50 @@ private static void copyHFileHalf(Configuration conf, Path inFile, Path outFile,
         .withChecksumType(StoreUtils.getChecksumType(conf))
         .withBytesPerCheckSum(StoreUtils.getBytesPerChecksum(conf)).withBlockSize(blocksize)
         .withDataBlockEncoding(familyDescriptor.getDataBlockEncoding()).withIncludesTags(true)
-        .withCreateTime(EnvironmentEdgeManager.currentTime()).build();
-      halfWriter = new StoreFileWriter.Builder(conf, cacheConf, fs).withFilePath(outFile)
-        .withBloomType(bloomFilterType).withFileContext(hFileContext).build();
+        .build();
+
       HFileScanner scanner = halfReader.getScanner(false, false, false);
       scanner.seekTo();
       do {
-        halfWriter.append(scanner.getCell());
+        final Cell cell = scanner.getCell();
+        if (null != halfWriter) {
+          halfWriter.append(cell);
+        } else {
+
+          // init halfwriter
+          if (conf.getBoolean(LOCALITY_SENSITIVE_CONF_KEY, DEFAULT_LOCALITY_SENSITIVE)) {
+            byte[] rowKey = CellUtil.cloneRow(cell);
+            HRegionLocation hRegionLocation = FutureUtils.get(loc.getRegionLocation(rowKey));
+            InetSocketAddress[] favoredNodes = null;
+            if (null == hRegionLocation) {
+              LOG.trace("Failed get of location, use default writer {}", Bytes.toString(rowKey));
```

Comment:
What does "default writer" mean here? A writer that doesn't take region location into account? Also, my impression here is that this is an unexpected behaviour. If that's the case, let's raise the log level to info or warn.

Reply:
yes,default writer that doesn't take region location into account,i will raise the log level to  warn